app/models/concerns/project_media_creators.rb

```diff
@@ -28,6 +30,66 @@ def create_annotation
     end
   end
 
+  def create_original_claim
+    claim = self.set_original_claim.strip
+    if claim.match?(/\A#{URI::DEFAULT_PARSER.make_regexp(['http', 'https'])}\z/)
+      uri = URI.parse(claim)
+      content_type = fetch_content_type(uri)
+
+      case content_type
+      when /^image\//
+        self.media = create_media_from_url('UploadedImage', claim)
+      when /^video\//
+        self.media = create_media_from_url('UploadedVideo', claim)
+      when /^audio\//
+        self.media = create_media_from_url('UploadedAudio', claim)
+      else
+        self.media = create_link_media(claim)
+      end
+    else
+      self.media = create_claim_media(claim)
+    end
+  end
+
+  def fetch_content_type(uri)
+    response = Net::HTTP.get_response(uri)
+    response['content-type']
+  end
+
+  def create_media_from_url(type, url)
+    klass = type.constantize
+    file = download_file(url)
+    m = klass.new
+    m.file = file
+    m.save
```

Comment:
Question: Any reason to not use `save!`? I prefer to raise an exception than failing silently.